src/renderer/inc/Cluster.hpp

```diff
@@ -16,24 +16,52 @@ Author(s):
 
 #pragma once
 
+#include <unicode.hpp>
+
 namespace Microsoft::Console::Render
 {
     class Cluster
     {
     public:
-        Cluster(const std::wstring_view text, const size_t columns);
+        constexpr Cluster() noexcept = default;

```

Comment:
At one point I realized that this entire class can be `constexpr` so I did that, because it removes a lot of code.
Also TIL: If you have `const` members in a class the class isn't copyable (since it can not be rewritten), which means that std::vector resizes are way less efficient. Now it's way faster without the const members.

Reply:
> TIL: If you have const members in a class the class isn't copyable (since it can not be rewritten), which means that std::vector resizes are way less efficient. 

This feels like a BIG DEAL.

Reply:
I mean it makes sense... kinda: A const member cannot be modified, and thus you also cannot have a default copy assignment operator anymore. And without that it also cannot be copied in bulk anymore (with memcpy). The only way to construct an instance now is using a placement-new in some uninitialized memory.

This creates some interesting situations around copying, like: https://godbolt.org/z/zdhdEY9s7